component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -161,58 +160,191 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                         .get(EmailOTPAuthenticatorConstants.IS_EMAILOTP_MANDATORY));
                 sendOtpToFederatedEmail = Boolean.parseBoolean(emailOTPParameters
                         .get(EmailOTPAuthenticatorConstants.SEND_OTP_TO_FEDERATED_EMAIL_ATTRIBUTE));
+                usecase = emailOTPParameters.get(EmailOTPAuthenticatorConstants.USE_CASE);
             } else {
                 isEmailOTPMandatory = Boolean.parseBoolean(String.valueOf(context.getProperty
                         (EmailOTPAuthenticatorConstants.IS_EMAILOTP_MANDATORY)));
                 sendOtpToFederatedEmail = Boolean.parseBoolean(String.valueOf(context.getProperty
                         (EmailOTPAuthenticatorConstants.SEND_OTP_TO_FEDERATED_EMAIL_ATTRIBUTE)));
+                usecase = (String) context.getProperty(EmailOTPAuthenticatorConstants.USE_CASE);
             }
-            FederatedAuthenticatorUtil.setUsernameFromFirstStep(context);
-            String username = String.valueOf(context.getProperty(EmailOTPAuthenticatorConstants.USER_NAME));
-            authenticatedUser = (AuthenticatedUser) context.getProperty
-                    (EmailOTPAuthenticatorConstants.AUTHENTICATED_USER);
-            // find the authenticated user.
-            if (authenticatedUser == null) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Cannot find the authenticated user, the username : " + username + " may be null");
-                }
-                throw new AuthenticationFailedException
-                        ("Authentication failed!. Cannot find the authenticated user, the username : "
-                                + username + " may be null");
-            }
-            boolean isUserExistence = FederatedAuthenticatorUtil.isUserExistInUserStore(username);
+
+            context.setProperty(EmailOTPAuthenticatorConstants.AUTHENTICATION,
+                    EmailOTPAuthenticatorConstants.AUTHENTICATOR_NAME);
             String queryParams = FrameworkUtils.getQueryStringWithFrameworkContextId(
                     context.getQueryParams(), context.getCallerSessionKey(),
                     context.getContextIdentifier());
-            if (isEmailOTPMandatory) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Process the EmailOTP mandatory flow ");
+            // If usecase property is not configured, email OTP will be executed in the below flow (Recommended flow)
```

Comment:
This code seems having too long methods from it's beginning. Can we break this down to methods with this effort so that it will be more readable :)

Reply:
done